### DISCUSSION TITLE
Initial lens model for DESIJ0618+5018

### DISCUSSION THREAD

**Discussion by ajmi-nishuti** [AstroBridge/BDLensing#164] [Q&A] [answered]
There are two arcs in normalized residuals. One of them has been reconstructed properly but there are residuals in lower arc. How can I model this properly? Is the masking good enough? 

Notebook Link : https://colab.research.google.com/drive/11MqPiYAjrNVYmtdabE-YfKjHklh83PeI?usp=sharing

![1](https://github.com/ajshajib/BDLensing/assets/132552905/754fdc97-58b1-4058-97ec-c24481831c6c)

**Comment by Rajit13** [comment]
Weird, lenstronomy made two separate arcs  (one blue and one red) where it was supposed to be 1 single arc at 3'o clock. I checked the values only the `gamma` value is high for you (comparing other models I've seen). Usually, 2 sersic profiles are enough for removing arc blue residuals, you've already done that.

**Reply by ajmi-nishuti** [reply]
![2](https://github.com/ajshajib/BDLensing/assets/132552905/15dc253c-b365-46ca-8e89-1a687c23924f)
![3](https://github.com/ajshajib/BDLensing/assets/132552905/bf5e1143-0e6c-44dd-8334-b060cca5fb2e)

**Reply by ajmi-nishuti** [reply]
After putting gamma = 1.5 (took a lower value as per your suggestion) ... the value of chi square decreased and the model turned out to be this.

**Reply by ajmi-nishuti** [reply]
Notebook link : https://colab.research.google.com/drive/1WF22MNbmAux7Cmsd77aBzxeb8E0zE9uc?usp=sharing

**Reply by Rajit13** [reply]
Ok, this model actually solves the double arc part. This current model is impressive. As Rafee Vai said, set a lower beta value. Change the `deltaPix_source`.

**Comment by rafee-adnan** [comment, marked as answer]
Follow these suggestions:
- [ ] do not use high `n_max` if not needed. Set `n_max` to 5
- [ ] try setting the upper limit of `beta` to 0.2 and inside source_sigma set `beta` to 0.01
- [ ] set the `deltaPix_source` to `0.01` inside `model_plot.source_plot()` as your reconstructed source is barely visible.

By the way, in ideal case your gamma value should be around 2. I think the above suggestions will fix the gamma issue.

**Reply by ajmi-nishuti** [reply]
bhai, should I work with two arc or the lower one ?

**Reply by rafee-adnan** [reply]
I think the current masking is fine. Go forward with it.

**Reply by ajmi-nishuti** [reply]
Bhai, I put gamma = 1.5 , then the lower blue residuals arc is gone.. Will you kindly tell me the reason ? 

https://colab.research.google.com/drive/1WF22MNbmAux7Cmsd77aBzxeb8E0zE9uc?usp=sharing

**Reply by rafee-adnan** [reply]
Notice that the `beta` value drastically gotten smaller the second time. Beta is the scale factor of shapelet. My assumption is if the `beta` value somehow converges to a big value, it tries to reconstruct bigger arcs, and it creates problem while fitting. That's why it helps when you keep the `beta` value smaller.

**Reply by ajmi-nishuti** [reply]
Then should I work with gamma=1.5 / one arc or the previous two arc model applying changes you've mentioned.

**Reply by rafee-adnan** [reply]
You can keep the lower limit of `gamma` 1.5, but you should initialize `gamma` at 2, as I have said before that in ideal case your gamma value should be around 2. Additionally, try following the other suggestions.

**Reply by ajmi-nishuti** [reply]
Bhai applying all the suggestions and putting the initial value of gamma to 1.9 , I got following result where gamma = 2.3.

Notebook link : https://colab.research.google.com/drive/1mYUdpFwBprKip_s-hZVy_SDUFJ6L3geK?usp=sharing

**Reply by rafee-adnan** [reply]
And I got this following the suggestions that I gave you. (notebook [here](https://drive.google.com/file/d/1Z4Nf5RZkwuFfV9V9QjAfFDcHmoZJe5z0/view?usp=sharing))
